File: cpp/year2019/day01_test.cc
#include "year2019/day01.h"

#include <fstream>
#include <sstream>

#include "gtest/gtest.h"

#include "adventofcode.h"

TEST(Year2019Day01, Part1Example1) {
  std::istringstream input("12");
  std::string output = "2";
  adventofcode::answer_t a = day01::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
}

TEST(Year2019Day01, Part1Example2) {
  std::istringstream input("14");
  std::string output = "2";
  adventofcode::answer_t a = day01::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
}

TEST(Year2019Day01, Part1Example3) {
  std::istringstream input("1969");
  std::string output = "654";
  adventofcode::answer_t a = day01::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
}

TEST(Year2019Day01, Part1Example4) {
  std::istringstream input("100756");
  std::string output = "33583";
  adventofcode::answer_t a = day01::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
}

TEST(Year2019Day01, Part1Actual) {
  std::ifstream input("year2019/testdata/01");
  std::string output = "3154112";
  adventofcode::answer_t a = day01::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

TEST(Year2019Day01, Part2Example1) {
  std::istringstream input("14");
  std::string output = "2";
  adventofcode::answer_t a = day01::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
}

TEST(Year2019Day01, Part2Example2) {
  std::istringstream input("1969");
  std::string output = "966";
  adventofcode::answer_t a = day01::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
}

TEST(Year2019Day01, Part2Example3) {
  std::istringstream input("100756");
  std::string output = "50346";
  adventofcode::answer_t a = day01::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
}

TEST(Year2019Day01, Part2Actual) {
  std::ifstream input("year2019/testdata/01");
  std::string output = "4728317";
  adventofcode::answer_t a = day01::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

File: java/src/test/java/com/github/saser/adventofcode/year2016/day01/Day01Test.java
package com.github.saser.adventofcode.year2016.day01;

import java.io.FileReader;
import java.io.IOException;
import java.io.StringReader;

import org.junit.Test;
import org.junit.Assert;

public class Day01Test {
    @Test
    public void part1Example1() {
        var input = new StringReader("R2, L3");
        var output = "5";
        var result = Day01.part1(input);
        Assert.assertEquals("no error", "", result.error);
        Assert.assertEquals("correct output", output, result.answer);
    }

    @Test
    public void part1Example2() {
        var input = new StringReader("R2, R2, R2");
        var output = "2";
        var result = Day01.part1(input);
        Assert.assertEquals("no error", "", result.error);
        Assert.assertEquals("correct output", output, result.answer);
    }

    @Test
    public void part1Example3() {
        var input = new StringReader("R5, L5, R5, R3");
        var output = "12";
        var result = Day01.part1(input);
        Assert.assertEquals("no error", "", result.error);
        Assert.assertEquals("correct output", output, result.answer);
    }

    @Test
    public void part1Actual() throws IOException {
        try (var input = new FileReader("inputs/2016/01")) {
            var output = "243";
            var result = Day01.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

    @Test
    public void part2Example() {
        var input = new StringReader("R8, R4, R4, R8");
        var output = "4";
        var result = Day01.part2(input);
        Assert.assertEquals("no error", "", result.error);
        Assert.assertEquals("correct output", output, result.answer);
    }

    @Test
    public void part2Actual() throws IOException {
        try (var input = new FileReader("inputs/2016/01")) {
            var output = "142";
            var result = Day01.part2(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }
}
